### PR TITLE
push SequenceNumber down from IFakeObjectCall to ICompletedFakeObjectCall

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

**Comment by adamralph** [comment]
This came out of digging around to investigate #651.

Whist it doesn't remove the property from the public API, I think it's a valid refactoring in its own right.

File: src/FakeItEasy/Core/CallWriter.cs [hunk 1]
```diff
@@ -17,7 +16,7 @@ public CallWriter(IFakeObjectCallFormatter callFormatter, IEqualityComparer<IFak
             this.callComparer = callComparer;
         }
 
-        public virtual void WriteCalls(IEnumerable<IFakeObjectCall> calls, IOutputWriter writer)
+        public virtual void WriteCalls<T>(IEnumerable<T> calls, IOutputWriter writer) where T : IFakeObjectCall
```
Comment (thomaslevesque) [inline src/FakeItEasy/Core/CallWriter.cs]: Is this a workaround for the lack of covariance in .NET 3.5 ? Clever :+1:

Comment (blairconrad) [inline src/FakeItEasy/Core/CallWriter.cs]: Thanks. I didn't even know I was working around something!

Comment (adamralph) [inline src/FakeItEasy/Core/CallWriter.cs]: @blairconrad does indeed deserve the credit for that one :smile:

**Comment by blairconrad** [comment]
Looks good to me.

**Comment by thomaslevesque** [comment]
> Looks good to me.

To me too

**Comment by blairconrad** [comment]
This issue has been fixed in [release 2.0.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.0.0).

**Pull request merged**